xcp_d/utils/confounds.py

```diff
@@ -245,21 +253,30 @@ def load_acompcor(confounds_df, confoundjs):
             # Pull out variance explained for CSF masks that are retained
             if value["Mask"] == "CSF" and value["Retained"]:
                 CSF.append([key, value["VarianceExplained"]])
-    # Select the first five components and add them to the list
-    csflist = []
-    wmlist = []
-    for i in range(0, 4):
-        try:
-            csflist.append(CSF[i][0])
-        except Exception as exc:
-            pass
-            print(exc)
-        try:
-            wmlist.append(WM[i][0])
-        except Exception as exc:
-            pass
-            print(exc)
-    acompcor = wmlist + csflist
+
+    # grab up to 5 acompcor values
+    N_COLS_TO_GRAB = 5
+    wm_comp_cor_retained = WM
+    csf_comp_cor_retained = CSF
+
+    # Note that column names were changed from a_comp_cor to w_comp_cor and c_comp_cor
+    # in later versions of fMRIPrep
+    n_wm_comp_cor = N_COLS_TO_GRAB
+    if len(wm_comp_cor_retained) < N_COLS_TO_GRAB:
+        LOGGER.warning(f"Only {len(wm_comp_cor_retained)} white matter CompCor columns found.")
+        n_wm_comp_cor = len(wm_comp_cor_retained)
+
+    # ditto for csf
+    n_csf_comp_cor = N_COLS_TO_GRAB
+    if len(csf_comp_cor_retained) < N_COLS_TO_GRAB:
+        LOGGER.warning(f"Only {len(wm_comp_cor_retained)} CSF CompCor columns found.")
+        n_wm_comp_cor = len(csf_comp_cor_retained)
+
+    acompcor_values = wm_comp_cor_retained[:n_wm_comp_cor] + csf_comp_cor_retained[:n_csf_comp_cor]
+    acompcor = []
+    for item in acompcor_values:
+        acompcor.append(item[0])
+
     return confounds_df[acompcor]
```

Comment:
```suggestion
    acompcor_columns = wm_comp_cor_retained[:n_wm_comp_cor] + csf_comp_cor_retained[:n_csf_comp_cor]

    return confounds_df[acompcor_columns]
```

Reply:
wm_comp_cor and csf_comp_cor are lists of lists with two items each

Reply:
We don't need the variance explained, so you can just drop that element.